source/val/validate_builtins.cpp

```diff
@@ -3123,16 +3123,6 @@ spv_result_t BuiltInsValidator::ValidateI32Vec4InputAtDefinition(
 
 spv_result_t BuiltInsValidator::ValidateWorkgroupSizeAtDefinition(
     const Decoration& decoration, const Instruction& inst) {
-  if (spvIsVulkanEnv(_.context()->target_env)) {
```

Comment:
@dneto0 

> the type of the constant is not checked. So someone could feed a bad module and still get a buffer overrund, I think.

In Vulkan we have the VUID
> The variable decorated with WorkgroupSize must be declared as a three-component vector of 32-bit integer values

I tried to write a Kernel shader but I can't think of how it could use a `Builtin WorkgroupSize` without an `ivec3` so I changed this to a global SPIR-V check